apps/io-wallet-user-func/src/infra/pdnd-interop/client.ts

```diff
@@ -0,0 +1,96 @@
+import { PdndInteropApiClientConfig } from "@/app/config";
+import * as TE from "fp-ts/lib/TaskEither";
+import { ServiceUnavailableError } from "io-wallet-common/error";
+import * as jwt from "jsonwebtoken";
+import { v4 as uuidv4 } from "uuid";
+
+import { VoucherRepository } from "../voucher";
+export class PdndInteropClient implements VoucherRepository {
+  static readonly CLIENT_ASSERTION_JWT_ALGORITHM = "RS256";
+  static readonly CLIENT_ASSERTION_JWT_TYPE = "JWT";
+  static readonly CLIENT_ASSERTION_TYPE =
+    "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
+  static readonly GRANT_TYPE = "client_credentials";
+  #audience: string;
+  #clientAssertionPrivateKey: string;
+  #clientId: string;
+  #kidId: string;
+  #purposeId: string;
+  #requestTimeout: number;
+  #url: string;
+  generateClientAssertion = () =>
+    jwt.sign(
+      {
```

Comment:
I haven't seen the whole PR yet, but `jsonwebtoken` shouldn't be used. In the whole codebase we are using [jose](https://www.npmjs.com/package/jose) which is a more up to date and easy to use library. Also check out the rest of the codebase on how to use it.